app/restaurant/controller.py

```diff
@@ -6,49 +6,221 @@
+        return jsonify({
+        'message' : 'Restaurant cannot be found',
+        'errors' : {
+        'restaurant' : 'No restaurant with id of %d' % restaurantId
+        }
+        }), 404
+
+    if request.method == 'GET': # single view
+        restaurant = restaurants[0].to_dict()
+        restaurant['owner'] = restaurants[0].owner.to_dict()
+
+        return jsonify({
+            'status' : 'success',
+            'message' : 'restaurant found',
             'data' : {
-                'restaurant' : restaurant.to_dict()
+                'restaurant' : restaurant
             }
-        }), 201
+        })
+    if request.method == 'DELETE': # single delete
+        if restaurants[0].owner.id != g.user['id'] and g.user['role'] != 'admin':
+            return jsonify({
+                'status' : 'error',
+                'message' : 'You are not allowed',
+                'errors' : {
+                    'user' : 'Not allowed'
+                }
+            }), 403
+
+
+        try:
+            db.session.delete(restaurants[0])
+            db.session.commit()
+        except:
+            return jsonify({
+                'status' : 'error',
+                'message' : 'There was a problem deleteing the restaurant',
+                'errors' : {
+                    'restaurant' : ['There was a problem deleteing the restaurant with id of %d' % restaurantId]
+                }
+            }), 500
 
-    return jsonify({
-        'message' : 'There is missing data',
-        'errors' : form.errors
-    })
+        return jsonify({
+            'status' : 'success',
+            'message' : 'The restaurant with id of %d was deleted.' % restaurantId
+        }), 200
+
+    else: # single update
+        if restaurants[0].owner.id != g.user['id'] and g.user['role'] != 'admin':
+            return jsonify({
+                'status' : 'error',
+                'message' : 'You are not allowed',
+                'errors' : {
+                    'user' : 'Not allowed'
+                }
+            }), 403
+
+        try:
+            request.json["restaurant_number"] = int(request.json["restaurant_number"])
+            print(request.json['restaurant_number'])
+        except:
+            return jsonify({
+                'status' : 'error',
+                'errors' : [
+                    'Restaurant number must be a positive integer.'
+                ],
+                'message' : 'There was a problem making the request.'
+            }), 400
 
+        request.json['owner_id'] = g.user['id']
 
-## view
+        form = RestaurantValidator(data=request.json)
 
+        if form.validate():
+            restaurant = restaurants[0]
 
-## update
+            restaurant.name = request.json['name']
+            restaurant.restaurant_number = request.json['restaurant_number']
+            restaurant.address = request.json['address']
 
+            # save the restaurant updates
+            db.session.commit()
 
```

Comment:
Ditto validation mentioned on line 96. 
The restaurant_number and/or address to be updated must not already be assigned to another Restaurant object in the system.